tests/test_optics.py

```diff
@@ -570,13 +543,20 @@ def test_OpticalPSF_pupil_plane():
     ref_psf = galsim.OpticalPSF(lam_over_diam, obscuration=obscuration, nstruts=nstruts,
                                 oversampling=pp_oversampling, strut_angle=strut_angle,
                                 pad_factor=pp_pad_factor)
-    im = galsim.fits.read(os.path.join(imgdir, pp_file))
+    if os.path.isfile(os.path.join(imgdir, pp_file)):
+        im = galsim.fits.read(os.path.join(imgdir, pp_file))
+    else:
+        im = galsim.Image(ref_psf._psf.aper.illuminated.astype(float))
+        im.write(os.path.join(imgdir, pp_file))
+    ref_im_scale = ref_psf._psf.aper.pupil_plane_scale
     test_psf = galsim.OpticalPSF(lam_over_diam, obscuration=obscuration,
                                  oversampling=pp_oversampling, pupil_plane_im=im,
-                                 pad_factor=pp_pad_factor)
+                                 pad_factor=pp_pad_factor,
+                                 _pupil_plane_scale=ref_im_scale)
```

Comment:
My intention was that this is something that users would normally not set directly.

The normal way to infer `Aperture.pupil_plane_scale` -- when generating the Aperture from an image and not geometric parameters -- is to look for the illuminated pixel farthest from the image center and declare that distance to be the radius.  You then divide by (half) the specified diameter in meters to get the scale.

Unfortunately, this is only accurate to ~1 pixel (see comment [here](https://github.com/GalSim-developers/GalSim/blob/%23549/galsim/phase_psf.py#L281)), since we can't know exactly where inside the pixel in question the edge of the pupil really is.  So _pupil_plane_scale lets you override this calculation if you have a more accurate value.  Actually, "more accurate" is probably not the right characterization, since the error is an inevitable consequence of discretizing the pupil into an array.  Really, the override is there so that its possible to create an `Aperture` parametrically, write the `illuminated` attribute to disk, and then recreate a byte-identical `Aperture` from the recorded image in this unit test.  I puzzled over how best to accomplish this for quite a while a few weeks ago.  If you have a better solution, I'm certainly open to it!